components/motor/gpio/motor_encoder_test.go

```diff
@@ -99,14 +98,23 @@ func TestMotorEncoder1(t *testing.T) {
 			test.That(tb, fakeMotor.PowerPct(), test.ShouldEqual, float32(1))
 		})
 
-		test.That(t, interrupt.Ticks(context.Background(), 99, nowNanosTest()), test.ShouldBeNil)
-		test.That(t, fakeMotor.Direction(), test.ShouldEqual, 1)
-
-		_motor.SetPower(context.Background(), .25, nil)
-		test.That(t, interrupt.Ticks(context.Background(), 1000, nowNanosTest()), test.ShouldBeNil) // go far!
+		errChan := make(chan error)
+		go func() {
+			ticksErr := interrupt.Ticks(context.Background(), 99, nowNanosTest())
+			errChan <- ticksErr
+		}()
+		time.Sleep(2 * time.Millisecond)
+		_motor.SetPower(context.Background(), -0.25, nil)
+		receivedErr := <-errChan
+		test.That(t, receivedErr, test.ShouldBeNil)
+		pos, err := _motor.Position(context.Background(), nil)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, fakeMotor.Direction(), test.ShouldEqual, -1)
+		test.That(t, pos, test.ShouldBeLessThan, 100)
```

Comment:
Does it have to be less than or can it be equal 99?

Reply:
The idea is that it should have interrupted the call with no guarantee as to what the exact position should be, but it definitely should not have gotten to 100. I'm going to change this though, I think GoFor should actually be returning an error if it gets cancelled and I should just be checking for that (although the test will no longer pass until your PR is merged)

Reply:
GoFor should not return an error on interruption. Let's not change this behavior in this PR.